In [1]:
import pandas as pd
import statsmodels.stats.api as sms
from scipy.stats import ttest_1samp, shapiro, levene, ttest_ind, mannwhitneyu, \
    pearsonr, spearmanr, kendalltau, f_oneway, kruskal
from statsmodels.stats.proportion import proportions_ztest

 İş Problemi: Kursun Büyük Çoğunluğunu İzleyenler ile İzlemeyenlerin Puanları Birbirinden Farklı mı?

 H0 : M1 = M2

 H1 : M1 != M2



In [9]:
df = pd.read_csv("course_reviews.csv")

# Rating: Kullanıcıların kursa verdiği puanları temsil eden sütun. 
# Timestamp: İnceleme veya verinin kaydedildiği zaman damgasını içeren sütun. 
# Enrolled: Kursa kaydolan kullanıcıların sayısını temsil eden sütun.
# Progress: Kullanıcıların kursun ne kadarını izlediğini yüzde olarak ifade eden sütun. 
# Questions Asked: Kullanıcıların kurs sırasında sordukları soru sayısını temsil eden sütun. 
# Questions Answered: Kurs eğitmeni tarafından yanıtlanan soruların sayısını içeren sütun. 


df.head()

Rating            Timestamp             Enrolled  Progress  \
0     5.0  2021-02-05 07:45:55  2021-01-25 15:12:08       5.0   
1     5.0  2021-02-04 21:05:32  2021-02-04 20:43:40       1.0   
2     4.5  2021-02-04 20:34:03  2019-07-04 23:23:27       1.0   
3     5.0  2021-02-04 16:56:28  2021-02-04 14:41:29      10.0   
4     4.0  2021-02-04 15:00:24  2020-10-13 03:10:07      10.0   

   Questions Asked  Questions Answered  
0              0.0                 0.0  
1              0.0                 0.0  
2              0.0                 0.0  
3              0.0                 0.0  
4              0.0                 0.0

In [4]:
# %75'ten fazlasını izleyenlerin rating ortalaması
df.loc[df["Progress"] > 75, "Rating"].mean()

4.860491071428571

In [5]:
# %25'ten azını izleyenlerin rating ortalaması
df.loc[df["Progress"] <= 25, "Rating"].mean()

4.730646306818182

Kursun çoğunu yani %75'inden fazlasını izlemiş kullanıcıların ortala verdiği puan, kursun az bir bölümünü yani %25'inden azını izlemiş kullanıcıların verdiği puanla farklı görünmüyor. Şimdi bu durum şans eseri mi ortaya çıkmış bir test edelim.

In [6]:
# Adım 1 : Hipotezleri kur
 # H0 : M1 = M2
 # H1 : M1 != M2

# Adım 2: Varsayımları test et

In [7]:
# Normallik varsayımı
# H0 : Kursun %75'inden fazlasını izlemiş kullanıcıların verdiği puanlar normal dağılıma uymaktadır.
# H1 : Kursun %75'inden fazlasını izlemiş kullanıcıların verdiği puanlar normal dağılıma uymamaktadır.

test_stat, pvalue = shapiro(df.loc[df["Progress"] > 75, "Rating"])

print("p_value :", pvalue)

if pvalue < 0.05:
    print("H0 reddedilir, örneklem normal dağılıma uymamaktadır")
else:
    print("H0 reddedilemez, örneklem normal dağılıma uymaktadır")

p_value : 1.1724449011561624e-37
H0 reddedilir, örneklem normal dağılıma uymamaktadır


Örneklem normal dağılıma uymadığından dolayı mannwhitneyu testini kullanırız.

In [8]:
test_stat, pvalue = mannwhitneyu(df.loc[df["Progress"] > 75, "Rating"],
                                  df.loc[df["Progress"] <= 25, "Rating"] )

print("p_value :", pvalue)

if pvalue < 0.05:
    print("H0 reddedilir, kursun büyük bir çoğunluğunu izleyenlerin rating ortalaması izlemeyenlerin ortalamasından farklıdır")
else:
    print("H0 reddedilemez, kursun büyük bir çoğunluğunu izleyenlerin rating ortalaması izlemeyenlerin ortalamasından farklı değildir")

p_value : 4.070866615096098e-10
H0 reddedilir, kursun büyük bir çoğunluğunu izleyenlerin rating ortalaması izlemeyenlerin ortalamasından farklıdır


%75'ten fazlasını izleyenlerin rating ortalaması : 4.860491071428571

%25'ten azını izleyenlerin rating ortalaması : 4.730646306818182

Rating ortalamaları birbirine yakın görünse de istatistiksel olarak aynı değildir. Sonuca göre, kursu daha çok izleyenlerin kursa daha yüksek puan verdiği sonucuna varabilirsiniz.